server/embedded/src/org/labkey/embedded/EmbeddedExtractor.java

```diff
@@ -162,7 +203,9 @@ public void extractDistribution(File webAppLocation)
         if (shouldExtract(webAppLocation))
         {
             labkeyWebappDirName = webAppLocation.getName();
+            File backupDir = createTempBackup(webAppLocation);
             extractExecutableJar(webAppLocation.getParentFile(), false);
+            deleteTempBackup(backupDir);
```

Comment:
Do this in a try/finally block? It'll use a random directory name, prefixed with `labkeyBackup`, so it's possible we could leak them if there are problems extracting from the JAR

Reply:
I was thinking that this way, the backup would be there if the extraction failed for some reason. That way they could, theoretically, recover to the point before attempting to upgrade. I can add some logging to help instruct folks in that direction.

Reply:
I'm curious what the recovery sequence would look like. Wouldn't they need to restore the previous JAR? And wouldn't it extract itself?